### PR TITLE
Delete instance + DayOfWeekGroup enum update

### PR DESCRIPTION
**Pull request by Brett09**
Issue No: #108 

- Event delete instance (and following instances if wanted)
- Added note that deleting instance and the following of an event will take time to be synced
- Properly retrieves the correct calendar
- Removed `Custom` item from `DayOfWeekGroup` enum

### CODE REVIEW

File: device_calendar/example/lib/presentation/pages/calendar_event.dart [hunk 1]
```diff
@@ -14,11 +15,14 @@ class CalendarEventPage extends StatefulWidget {
   final Calendar _calendar;
   final Event _event;
 
-  CalendarEventPage(this._calendar, [this._event]);
+  final VoidCallback _onLoadingStarted;
+  final Function(bool) _onDeleteFinished;
+
+  CalendarEventPage(this._calendar, [this._event, this._onLoadingStarted, this._onDeleteFinished]);
```
Comment (nickrandolph) [inline device_calendar/example/lib/presentation/pages/calendar_event.dart]: why are we passing callbacks into the page constructor - this doesn't seem like a good idea.

Comment (Brett09) [reply device_calendar/example/lib/presentation/pages/calendar_event.dart]: Removed it and now passes a widget instead

File: device_calendar/android/src/main/kotlin/com/builttoroam/devicecalendar/CalendarDelegate.kt [hunk 2]
```diff
@@ -449,9 +455,76 @@ class CalendarDelegate : PluginRegistry.RequestPermissionsResultListener {
...
+                val deleteSucceeded = contentResolver?.delete(eventsUriWithId, null, null) ?: 0
+                finishWithSuccess(deleteSucceeded > 0, pendingChannelResult)
+            }
+            else {
+                if (!followingInstances!!) { // Only this instance
+                    val exceptionUriWithId = ContentUris.withAppendedId(Events.CONTENT_EXCEPTION_URI, eventIdNumber)
+                    val values = ContentValues()
+                    val instanceCursor = CalendarContract.Instances.query(contentResolver, EVENT_INSTANCE_DELETION, startDate!!, endDate!!)
+
+                    for (x in 0 until instanceCursor.count) {
```
Comment (nickrandolph) [inline device_calendar/android/src/main/kotlin/com/builttoroam/devicecalendar/CalendarDelegate.kt]: change this to use "while(instanceCursor.moveNext())"

File: device_calendar/android/src/main/kotlin/com/builttoroam/devicecalendar/CalendarDelegate.kt [hunk 3]
```diff
@@ -449,9 +455,76 @@ class CalendarDelegate : PluginRegistry.RequestPermissionsResultListener {
...
+                    val deleteSucceeded = contentResolver?.insert(exceptionUriWithId, values)
+                    finishWithSuccess(deleteSucceeded != null, pendingChannelResult)
+                }
+                else { // This and following instances
+                    val eventsUriWithId = ContentUris.withAppendedId(Events.CONTENT_URI, eventIdNumber)
+                    val values = ContentValues()
+                    val instanceCursor = CalendarContract.Instances.query(contentResolver, EVENT_INSTANCE_DELETION, startDate!!, endDate!!)
+
+                    for (x in 0 until instanceCursor.count) {
+                        instanceCursor.moveToNext()
```
Comment (nickrandolph) [inline device_calendar/android/src/main/kotlin/com/builttoroam/devicecalendar/CalendarDelegate.kt]: same as previous comment - use while loop

File: device_calendar/android/src/main/kotlin/com/builttoroam/devicecalendar/CalendarDelegate.kt [hunk 4]
```diff
@@ -449,9 +455,76 @@ class CalendarDelegate : PluginRegistry.RequestPermissionsResultListener {
...
+                                for (y in 0 until cursor.count) {
+                                    cursor.moveToNext()
+                                    if (eventIdNumber == cursor.getLong(EVENT_INSTANCE_DELETION_ID_INDEX)) {
+                                        newRule.count--
+                                    }
+                                }
+                                cursor.close()
+                            }
+                            else { // Indefinite and specified date rule
+                                val cursor = CalendarContract.Instances.query(contentResolver, EVENT_INSTANCE_DELETION, 946645200000, startDate - 1) // 946645200000 = 01/01/2000
```
Comment (nickrandolph) [inline device_calendar/android/src/main/kotlin/com/builttoroam/devicecalendar/CalendarDelegate.kt]: change constant - to long max

File: device_calendar/lib/src/common/calendar_enums.dart [hunk 5]
```diff
@@ -12,8 +12,7 @@ enum DayOfWeekGroup {
   None,
   Weekday,
   Weekend,
-  Alldays,
-  Custom
+  Alldays
```
Comment (nickrandolph) [inline device_calendar/lib/src/common/calendar_enums.dart]: AllDays

File: device_calendar/android/src/main/kotlin/com/builttoroam/devicecalendar/CalendarDelegate.kt [hunk 6]
```diff
@@ -449,9 +455,76 @@ class CalendarDelegate : PluginRegistry.RequestPermissionsResultListener {
...
+                                var lastRecurrenceDate = 0.toLong()
+
+                                for (y in 0 until cursor.count) {
+                                    cursor.moveToNext()
+                                    if (eventIdNumber == cursor.getLong(EVENT_INSTANCE_DELETION_ID_INDEX)) {
+                                        lastRecurrenceDate = cursor.getLong(EVENT_INSTANCE_DELETION_END_INDEX)
+                                    }
+                                }
+
+                                newRule.until = DateTime(lastRecurrenceDate)
```
Comment (nickrandolph) [inline device_calendar/android/src/main/kotlin/com/builttoroam/devicecalendar/CalendarDelegate.kt]: use startdate-1 if lastRecurrenceDate is null

File: device_calendar/android/src/main/kotlin/com/builttoroam/devicecalendar/CalendarDelegate.kt [hunk 7]
```diff
@@ -449,9 +455,76 @@ class CalendarDelegate : PluginRegistry.RequestPermissionsResultListener {
...
+                                        newRule.count--
+                                    }
+                                }
+                                cursor.close()
+                            }
+                            else { // Indefinite and specified date rule
+                                val cursor = CalendarContract.Instances.query(contentResolver, EVENT_INSTANCE_DELETION, 946645200000, startDate - 1) // 946645200000 = 01/01/2000
+                                var lastRecurrenceDate = 0.toLong()
+
+                                for (y in 0 until cursor.count) {
```
Comment (nickrandolph) [inline device_calendar/android/src/main/kotlin/com/builttoroam/devicecalendar/CalendarDelegate.kt]: Use while loop

File: device_calendar/android/src/main/kotlin/com/builttoroam/devicecalendar/CalendarDelegate.kt (hunk 2 above)
Comment (Brett09) [reply device_calendar/android/src/main/kotlin/com/builttoroam/devicecalendar/CalendarDelegate.kt]: Updated all loops

File: device_calendar/android/src/main/kotlin/com/builttoroam/devicecalendar/CalendarDelegate.kt (hunk 4 above)
Comment (Brett09) [reply device_calendar/android/src/main/kotlin/com/builttoroam/devicecalendar/CalendarDelegate.kt]: Updated (using constants: `startDate - DateUtils.YEAR_IN_MILLIS`)

File: device_calendar/android/src/main/kotlin/com/builttoroam/devicecalendar/CalendarDelegate.kt (hunk 6 above)
Comment (Brett09) [reply device_calendar/android/src/main/kotlin/com/builttoroam/devicecalendar/CalendarDelegate.kt]: Updated to be a null value and added null check

**Pull request merged**